pkg/model/components/defaults.go

```diff
@@ -37,7 +37,11 @@ func (b *DefaultsOptionsBuilder) BuildOptions(o interface{}) error {
 	}
 
 	if options.ContainerRuntime == "" {
-		options.ContainerRuntime = "docker"
+		if b.Context.IsKubernetesGTE("1.20") {
```

Comment:
Should also check that `options.Docker` is not set, otherwise we will probably break the cluster.

Reply:
My new code is not quite working as expected during the kops `edit` and `update` and needs some further debugging.

```
	if options.ContainerRuntime == "" {
		if b.Context.IsKubernetesGTE("1.20") {
			options.ContainerRuntime = "containerd"

			if options.Docker != nil {
				klog.Errorf("Docker options are invalid with the default container runtime, containerd")
			}
		} else {
			options.ContainerRuntime = "docker"
		}
	}
```

Reply:
@hakman 
It's working as expected, I've forgotten to remove `ContainerRuntime: containerd` from the cluster spec.

```
$ kops edit cluster --state ${AWS_KOPS_STATE_STORE} --name ${KOPS_CLUSTER_NAME}
E0113 22:24:30.445745   24928 defaults.go:45] Docker options are invalid with the default container runtime, containerd
```